models/perpetuals/optimism/pika_v2_optimism.sql

```diff
@@ -0,0 +1,161 @@
+		,evt_block_number
+		,'2' AS version
+	FROM {{ source('pika_perp_v2_optimism', 'PikaPerpV2_evt_ClosePosition') }}
+	{% if is_incremental() %}
+	WHERE evt_block_time >= DATE_TRUNC("DAY", NOW() - INTERVAL '1 WEEK')
+	{% endif %}
+),
+
+perps AS (
+	SELECT
+		evt_block_time AS block_time
+		
+		,CASE
+		WHEN productId = 1 OR productId = 16 THEN 'ETH'
+		WHEN productId = 2 OR productId = 17 THEN 'BTC'
+		WHEN productId = 3 OR productId = 18 THEN 'LINK'
+		WHEN productId = 4 OR productId = 19 THEN 'SNX'
+		WHEN productId = 5 OR productId = 20 THEN 'SOL'
+		WHEN productId = 6 OR productId = 21 THEN 'AVAX'
+		WHEN productId = 7 OR productId = 22  THEN 'MATIC'
+		WHEN productId = 8 THEN 'LUNA'
+		WHEN productId = 9 OR productId = 23 THEN 'AAVE'
+		WHEN productId = 10 OR productId = 24 THEN 'APE'
+		WHEN productId = 11 OR productId = 25 THEN 'AXS'
+		WHEN productId = 12 OR productId = 26 THEN 'UNI'
+		ELSE CONCAT ('product_id_', productId)
+		END AS virtual_asset
+
+		,CASE
+		WHEN productId = 1 OR productId = 16 THEN 'ETH'
+		WHEN productId = 2 OR productId = 17 THEN 'BTC'
+		WHEN productId = 3 OR productId = 18 THEN 'LINK'
+		WHEN productId = 4 OR productId = 19 THEN 'SNX'
+		WHEN productId = 5 OR productId = 20 THEN 'SOL'
+		WHEN productId = 6 OR productId = 21 THEN 'AVAX'
+		WHEN productId = 7 OR productId = 22  THEN 'MATIC'
+		WHEN productId = 8 THEN 'LUNA'
+		WHEN productId = 9 OR productId = 23 THEN 'AAVE'
+		WHEN productId = 10 OR productId = 24 THEN 'APE'
+		WHEN productId = 11 OR productId = 25 THEN 'AXS'
+		WHEN productId = 12 OR productId = 26 THEN 'UNI'
+		ELSE CONCAT ('product_id_', productId)
+		END AS underlying_asset
+
+		,CASE
+		WHEN productId = 1 OR productId = 16 THEN 'ETH-USD'
+		WHEN productId = 2 OR productId = 17 THEN 'BTC-USD'
+		WHEN productId = 3 OR productId = 18 THEN 'LINK-USD'
+		WHEN productId = 4 OR productId = 19 THEN 'SNX-USD'
+		WHEN productId = 5 OR productId = 20 THEN 'SOL-USD'
+		WHEN productId = 6 OR productId = 21 THEN 'AVAX-USD'
+		WHEN productId = 7 OR productId = 22  THEN 'MATIC-USD'
+		WHEN productId = 8 THEN 'LUNA-USD'
+		WHEN productId = 9 OR productId = 23 THEN 'AAVE-USD'
+		WHEN productId = 10 OR productId = 24 THEN 'APE-USD'
+		WHEN productId = 11 OR productId = 25 THEN 'AXS-USD'
+		WHEN productId = 12 OR productId = 26 THEN 'UNI-USD'
+		ELSE CONCAT ('product_id_', productId)
+		END AS market
+		
+		,contract_address AS market_address
+		,(margin/1e8) * (leverage/1e8) AS volume_usd
+		,fee/1e8 AS fee_usd
+		,margin/1e8 AS margin_usd
+
+		,CASE
+		WHEN isLong = 'true' THEN 'long'
+		WHEN isLong = 'false' THEN 'short'
+		ELSE isLong
+		END AS trade
+
+		,'Pika' AS project
+		,version
+		,user AS trader
+		,margin * leverage AS volume_raw
+		,evt_tx_hash AS tx_hash
+		,evt_index
+	FROM positions
+	{% if is_incremental() %}
+	WHERE evt_block_time >= DATE_TRUNC("DAY", NOW() - INTERVAL '1 WEEK')
+	{% endif %}
+
+)
+
+SELECT
+	'optimism' AS blockchain
+	,perps.block_time
+	,perps.virtual_asset
```

Comment:
it looks like this column doesn't exist

Reply:
> it looks like this column doesn't exist

@soispoke which column are you referring to? Or did you mean all three columns?

> If this is confusing let me know and I can commit some changes myself, happy to help out

Thanks! I got the hang of it though and made the changes. Will be ready to commit once I get the clarity on which column/s you were referring to above. :)

Reply:
@RplusT Was referring specifically to the `perps.virtual_asset` column

Reply:
@soispoke that's odd, it's running fine when I try out the compiled code on dune.com: https://dune.com/queries/1243365?d=1

any suggestion on where else I should look to find a fix?

Reply:
@RplusT I think you're just not getting the `virtual_asset` values from your `perps` CTE ?

Reply:
@soispoke the `perps` CTE derives it from a `CASE` statement involving the `productId` values from `pika_perp_v2_optimism.PikaPerpV2_evt_NewPosition` and `pika_perp_v2_optimism.PikaPerpV2_evt_ClosePosition`. 

```
,CASE
		WHEN productId = 1 OR productId = 16 THEN 'ETH'
		WHEN productId = 2 OR productId = 17 THEN 'BTC'
		WHEN productId = 3 OR productId = 18 THEN 'LINK'
		WHEN productId = 4 OR productId = 19 THEN 'SNX'
		WHEN productId = 5 OR productId = 20 THEN 'SOL'
		WHEN productId = 6 OR productId = 21 THEN 'AVAX'
		WHEN productId = 7 OR productId = 22  THEN 'MATIC'
		WHEN productId = 8 THEN 'LUNA'
		WHEN productId = 9 OR productId = 23 THEN 'AAVE'
		WHEN productId = 10 OR productId = 24 THEN 'APE'
		WHEN productId = 11 OR productId = 25 THEN 'AXS'
		WHEN productId = 12 OR productId = 26 THEN 'UNI'
		ELSE CONCAT ('product_id_', productId)
		END AS virtual_asset
```

Reply:
started new comment below on correct file